packages/instruction-plans/src/transaction-plan-executor.ts

```diff
@@ -0,0 +1,7 @@
+import type { TransactionPlan } from './transaction-plan';
+import type { TransactionPlanResult } from './transaction-plan-result';
+
+export type TransactionPlanExecutor<TContext extends object = object> = (
```

Comment:
Just checking that you meant `object` (eg. `WritableStream` would be allowable here) rather than `Record<number | string | symbol, unknown>`?